### ISSUE TITLE
ECONNRESET on /check_proxy and /phpmyadmin URLs

### ISSUE THREAD

**Issue by stevencohn** [indexzero/http-server#475]
We have a public-facing Web site that we use for demos, not production. And we're using http-server for simplicity. However, noticing that we're seeing incoming HEAD requests with URLs of the form /check_proxy and /phpmyadmin, mostly from suspicious IP addresses.

But the HEAD request causes http-server to crash as shown below. This is easily reproducible by simulating a request using Postman.

We're switching this server to use nginx instead which is more resilient but thought it was worth mentioning if there was a possibility of a fix... Thanks!

```
C:\Users\layer1builder\AppData\Roaming\npm\node_modules\http-server\node_modules\http-proxy\lib\http-proxy\index.js:120
throw err;
^
Error: read ECONNRESET
at _errnoException (util.js:999:13)
at TCP.onread (net.js:629:25)
```

**Comment by BigBlueHat** [comment]
@stevencohn thanks for reporting this situation. What version of http-server were you using and with what command line parameters? A simple HEAD request (even one that 404's) should not have crashed http-server (obviously... 😜)--meaning...I've not seen nor heard of this happening. More details would help us debug it. Thanks!

**Comment by stevencohn** [comment]
Can be reproduced with the latest version and the command line:
http-server -p 80 -P http://localhost:9000
I think it has to do with the proxy redirect parameter; don't think it can be repo without it.

**Comment by thornjad** [comment]
This may be fixed by #477, please re-open if this is not the case.
